### PR TITLE
[Iga] Nurbs Volume Test

### PR DESCRIPTION
**Pull request by manuelmessmer**
**Description**
Adding test for IGA volume elements.
The corresponding nurbs volume geometries were added in #7742.

### CODE REVIEW

File: applications/IgaApplication/tests/test_IgaApplication.py [hunk 1]
```diff
@@ -31,6 +33,7 @@ def AssembleTestSuites():
         ScordelisRoofShell3pTest,
         TTestModelers
         ]))
+    smallSuite.addTests(KratosUnittest.TestLoader().loadTestsFromTestCases([TTestNurbsVolumeElements]))
```
Comment (tteschemacher) [inline applications/IgaApplication/tests/test_IgaApplication.py]:
```suggestion
        ScordelisRoofShell3pTest,
        TTestNurbsVolumeElements,
        TTestModelers
        ]))
```

I reckon it is more readable in that way.

File: applications/IgaApplication/tests/test_nurbs_volume_element.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
+import KratosMultiphysics as KM
+from KratosMultiphysics.IgaApplication import *
+
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+def drange(start, stop, step):
```
Comment (tteschemacher) [inline applications/IgaApplication/tests/test_nurbs_volume_element.py]: that's a cool solution. Just the name is a bit confusing.

**Comment by tteschemacher** [comment]
It is pretty awesome that this works. I only have really minor comments. Other than those, I am fine with it. Thanks!!

**Comment by tteschemacher** [comment]
one of the codacy suggestions might also be nice.

File: applications/IgaApplication/tests/test_nurbs_volume_element.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+        # Create a nurbs volume
+        volume = TestNurbsVolumeElement.create_geometry(model_part, order_u, order_v_w, length)
+
+        # Create quadrature_point_geometries
+        quadrature_point_geometries = KM.GeometriesVector()
+
+        volume.CreateQuadraturePointGeometries(quadrature_point_geometries, 2)
+
+        for i in range(len(quadrature_point_geometries)):
```
Comment (manuelmessmer) [inline applications/IgaApplication/tests/test_nurbs_volume_element.py:87]: For some reason the enumerate or a standard for loop on the quadrature point geometries does not work. There is an error with the ownership between python and C++. To make it work I would have to change the 'self.begin()' to 'self.ptr_begin()' in the following line....

https://github.com/KratosMultiphysics/Kratos/blob/4216e1312c737825b58acd60e40abd90227762c0/kratos/python/containers_interface.h#L94

**Comment by manuelmessmer** [comment]
@tteschemacher  thanks for reviewing.

File: applications/IgaApplication/tests/test_nurbs_volume_element.py (hunk 3 above)
Comment (tteschemacher) [reply applications/IgaApplication/tests/test_nurbs_volume_element.py:87]: Ok. I've never used it myself but I found it interesting. Nevertheless that's absolutely minor.

**Review by tteschemacher — APPROVED** [review]
Awesome 🦘🦈

**Pull request merged**